admin/product.php

```diff
@@ -20,9 +20,11 @@
     $type_handler = $product_lookup->getTypeHandler() . '.php';
 }
 
-if ($product_lookup === null || !$product_lookup->exists()) {
-    $messageStack->add_session(sprintf(WARNING_PRODUCT_DOES_NOT_EXIST, (int)($_GET['pID'] ?? 0)), 'warning');
-    zen_redirect(zen_href_link(FILENAME_CATEGORY_PRODUCT_LISTING));
+if ($action !== 'new_product' && $action !== 'new_product_preview' && $action !== 'insert_product') { 
```

Comment:
from a readability standpoint, i think `in_array` is better than these multiple conditionals.